mathesar_ui/src/stores/table-data/records.ts

```diff
@@ -461,19 +462,28 @@ export class RecordsData {
 
     let shouldReFetchRecords = successRowKeys.size > 0;
     if (primaryKeysOfSavedRows.length > 0) {
-      // TODO: Convert this to single request
-      const promises = primaryKeysOfSavedRows.map((pk) =>
-        deleteAPI<RowKey>(`${this.url}${pk}/`)
-          .then(() => {
-            successRowKeys.add(pk);
-            return successRowKeys;
-          })
-          .catch((error: unknown) => {
-            failures.set(pk, getErrorMessage(error));
-            return failures;
-          }),
-      );
-      await Promise.all(promises);
+      // Converted delete to single bulk delete request as per https://github.com/centerofci/mathesar/issues/2701
+      const recordIds: number[] = [];
+      primaryKeysOfSavedRows.map((n) => {
+        recordIds.push(Number(n));
+        return Number(n);
+      });
```

Comment:
Primary keys are not necessarily numbers, they can be of any type. In cases where we are connecting to an existing user database or tables which are created from outside Mathesar, the pk could be of a different type.